gravitee-io/gravitee-access-management
# fix: create & use updatePassword on UserProvider

fixes gravitee-io/issues#7800

## :id: Reference related issue. 


## :pencil2: A description of the changes proposed in the pull request

Purpose of this change is to provide dedicated method in the UserProvider interface to update user password instead of relying on a single update user method.

## :memo: Test scenarios 

You need to do these tests with Mongo IDP, JBDBC IDP and HTTP IDP.
:warning: For JDBC & Mongo, test a configuration with and withour dedicated salf field

* Update a user attribute through UI ==> password and salt are not changed
* Update a user attribute through SCIM (do not include the password field) ==> password and salt are not changed

* Update a user password through UI ==> password and salt are changed
* Update a user attribute through SCIM by including the password field ==> password and salt are changed

* Request a new password on the GW using forgot password link ==> after the reset password workflow, password and salt are changed

**NOTE:** after each point, try to connect the end user on the GW to confirm that the password has changed or not according to the case
